test/fuzz/sidecarset.go

```diff
@@ -0,0 +1,326 @@
+	return nil
+}
+
+func GenerateSidecarSetNamespace(cf *fuzz.ConsumeFuzzer, sidecarSet *appsv1alpha1.SidecarSet) error {
+	isStructured, err := cf.GetBool()
+	if err != nil {
+		return err
+	}
+
+	if !isStructured {
+		namespace, err := cf.GetString()
+		if err != nil {
+			return err
+		}
+		sidecarSet.Spec.Namespace = namespace
+		return nil
+	}
+
+	if valid, err := cf.GetBool(); valid && err == nil {
+		sidecarSet.Spec.Namespace = GenerateValidNamespaceName()
+	} else {
+		sidecarSet.Spec.Namespace = GenerateInvalidNamespaceName()
+	}
+
+	return nil
+}
+
+func GenerateSidecarSetNamespaceSelector(cf *fuzz.ConsumeFuzzer, sidecarSet *appsv1alpha1.SidecarSet) error {
+	selector := &metav1.LabelSelector{}
+	if err := GenerateLabelSelector(cf, selector); err != nil {
+		return err
+	}
+	sidecarSet.Spec.NamespaceSelector = selector
+	return nil
+}
+
+func GenerateSidecarSetInitContainer(cf *fuzz.ConsumeFuzzer, sidecarSet *appsv1alpha1.SidecarSet) error {
+	isStructured, err := cf.GetBool()
+	if err != nil {
+		return err
+	}
+
+	if !isStructured {
+		initContainers := make([]appsv1alpha1.SidecarContainer, 0)
+		if err := cf.CreateSlice(&initContainers); err != nil {
+			return err
+		}
+		sidecarSet.Spec.InitContainers = initContainers
+		return nil
+	}
+
+	for i := range sidecarSet.Spec.InitContainers {
+		if err := GenerateSidecarContainer(cf, &sidecarSet.Spec.InitContainers[i]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func GenerateSidecarSetContainer(cf *fuzz.ConsumeFuzzer, sidecarSet *appsv1alpha1.SidecarSet) error {
+	isStructured, err := cf.GetBool()
+	if err != nil {
+		return err
+	}
+
+	if !isStructured {
+		containers := make([]appsv1alpha1.SidecarContainer, 0)
+		if err := cf.CreateSlice(&containers); err != nil {
+			return err
+		}
+		sidecarSet.Spec.Containers = containers
+		return nil
+	}
+
+	for i := range sidecarSet.Spec.Containers {
+		if err := GenerateSidecarContainer(cf, &sidecarSet.Spec.Containers[i]); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func GenerateSidecarContainer(cf *fuzz.ConsumeFuzzer, sidecarContainer *appsv1alpha1.SidecarContainer) error {
+	isStructured, err := cf.GetBool()
+	if err != nil {
+		return err
+	}
+
+	if err := cf.GenerateStruct(sidecarContainer); err != nil {
+		return err
+	}
+
+	if !isStructured {
+		return nil
+	}
+
+	choice, err := cf.GetInt()
+	if err != nil {
+		return err
+	}
+
+	validPodInjectPolicyType := []appsv1alpha1.PodInjectPolicyType{
+		appsv1alpha1.BeforeAppContainerType,
+		appsv1alpha1.AfterAppContainerType,
+	}
+
+	validSidecarContainerUpgradeType := []appsv1alpha1.SidecarContainerUpgradeType{
+		appsv1alpha1.SidecarContainerColdUpgrade,
+		appsv1alpha1.SidecarContainerHotUpgrade,
+	}
+
+	validShareVolumePolicy := []appsv1alpha1.ShareVolumePolicyType{
+		appsv1alpha1.ShareVolumePolicyEnabled,
+		appsv1alpha1.ShareVolumePolicyDisabled,
+	}
+	sidecarContainer.PodInjectPolicy = validPodInjectPolicyType[choice%(len(validPodInjectPolicyType))]
```

Comment:
we can also shuffle the restart policy of init container, which will generate cases for native sidecar

Reply:
> we can also shuffle the restart policy of init container, which will generate cases for native sidecar

done